packages/oo-ascii-tree/README.md

```diff
@@ -0,0 +1,166 @@
+# oo-ascii-tree
+
+Renders ASCII trees from an object-oriented object graph.
+
+Features:
+
+ * Multiline text
+ * Multiple root nodes
+
+Roadmap:
+
+ * Customization of tree formatting and indentation size
+
+## Install
+
+```console
+$ npm i oo-ascii-tree
+```
+
+## Basic Example
+
+```ts
+import { AsciiTree } from '../lib';
+
+const tree = new AsciiTree('root');
+
+tree.add(new AsciiTree('child1'));
+
+tree.add(new AsciiTree('child2',
+  new AsciiTree('grandchild1'),
+  new AsciiTree('grandchild2')
+));
+
+tree.add(new AsciiTree('child3'));
+
+tree.printTree();
+```
+
+Prints the following tree to stdout:
+
+```
+root
+ ├── child1
+ ├─┬ child2
+ │ ├── grandchild1
+ │ └── grandchild2
+ └── child3
+```
+
+## Advanced Usage
+
+You can also subclass `AsciiTree` to encapsulate some model. The following
+example declares a `TitleNode` which formats a title with "====" underline and a
```

Comment:
This is good, but there's no actual need to subclass right?

Reply:
It's a useful paradigm for certain use cases. Not required.